acceptancetests/assess_network_spaces.py

```diff
@@ -0,0 +1,600 @@
+#!/usr/bin/env python
+"""Assess network spaces for supported providers (currently only EC2)"""
+
+import argparse
+import logging
+import sys
+import json
+import yaml
+import subprocess
+import re
+import os
+import ipaddress
+import boto3
+
+from jujupy import (
+    client_from_config,
+    client_for_existing
+    )
+from jujupy.exceptions import (
+    ProvisioningError
+    )
+from deploy_stack import (
+    BootstrapManager
+    )
+from utility import (
+    add_basic_testing_arguments,
+    generate_default_clean_dir,
+    configure_logging,
+    )
+
+__metaclass__ = type
+
+log = logging.getLogger("assess_network_spaces")
+
+
+class AssessNetworkSpaces:
+
+    def __init__(self, args):
+        if args.logs:
+            self.log_dir = args.logs
+        else:
+            self.log_dir = generate_default_clean_dir(
+                            args.temp_env_name)
+
+    def assess_network_spaces(self, client, target_model=None, series=None):
+        """Assesses network spaces
+
+        :param client: The juju client in use
+        :param target_model: Optional existing model to test under
+        :param series: Ubuntu series to deploy
+        """
+        self.setup_testing_environment(client, target_model, series)
+        log.info('Starting network tests.')
+        failures, failmsg = self.testing_iterations(client)
+        if failures:
+            raise TestFailure('Test failures:' + failmsg)
+        log.info('SUCESS')
+        return
+
+
+    def testing_iterations(self, client):
+        """Verify that spaces are set up proper and functioning
+
+        :param client: Juju client object with machines and spaces
+        """
+        failures = 0
+        failmsg = ''
+        alltests = [
+            self.verify_machine_spaces,
```

Comment:
can we prefix these methods with test_ or similar to denote they are the testing calls.

Reply:
I renamed a couple of the tests and prepended assert_.